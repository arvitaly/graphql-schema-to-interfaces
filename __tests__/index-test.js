"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = require("./..");
const interfaces_1 = require("./../__fixtures__/interfaces");
const schema_1 = require("./../__fixtures__/schema");
describe("Convert", () => {
    it("simple", () => {
        expect(__1.default(schema_1.default)).toBe(interfaces_1.default);
    });
});
