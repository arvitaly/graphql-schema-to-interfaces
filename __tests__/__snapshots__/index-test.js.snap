// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert simple 1`] = `
"export interface IQuery {
    node?(params: IQuerynodeParams): any;
    viewer?: IViewer;
}
export interface IViewer {
    id: string;
    model1?(params?: IViewermodel1Params): IModel1Connection;
}
export interface IViewermodel1Params {
    after?: string;
    first?: number;
    before?: string;
    last?: number;
}
export interface IModel1Connection {
    edges: IModel1ConnectionEdge[];
    pageInfo?: IModel1ConnectionPageInfo;
}
export interface IModel1ConnectionPageInfo {
    hasNextPage?: boolean;
}
export interface IModel1ConnectionEdge {
    node?: IModel1;
}
export interface IModel1 {
    id: string;
    field1?: string;
    model2?(params: IModel1model2Params): IModel2;
}
export interface IModel1model2Params {
    where: IWhere;
}
export interface IWhere {
    field1: string;
    fieldx?: number;
}
export interface IModel2 {
    field2?: number;
    id: string;
}
export interface IQuerynodeParams {
    id: string;
}"
`;

exports[`Convert simple 2`] = `
"export interface IQuery {
    node(params: IQuerynodeParams): any;
    viewer: IViewer;
}
export interface IViewer {
    id: string;
    model1(params?: IViewermodel1Params): IModel1Connection;
}
export interface IViewermodel1Params {
    after?: string;
    first?: number;
    before?: string;
    last?: number;
}
export interface IModel1Connection {
    edges: IModel1ConnectionEdge[];
    pageInfo: IModel1ConnectionPageInfo;
}
export interface IModel1ConnectionPageInfo {
    hasNextPage: boolean;
}
export interface IModel1ConnectionEdge {
    node: IModel1;
}
export interface IModel1 {
    id: string;
    field1: string;
    model2(params: IModel1model2Params): IModel2;
}
export interface IModel1model2Params {
    where: IWhere;
}
export interface IWhere {
    field1: string;
    fieldx?: number;
}
export interface IModel2 {
    field2: number;
    id: string;
}
export interface IQuerynodeParams {
    id: string;
}"
`;
